### DISCUSSION TITLE
How to access `context.cwd`

### DISCUSSION THREAD

**Discussion by Mister-Hope** [vuepress/core#43] [Q&A] [answered]
A new question here,

I found:

- `context.isProd` change to ` app.env.isProd`
- `context.outDir` change to `app.options.dest`
- `context.base` change to `app.options.base`
- `context.sourceDir` changed to `app.options.source`

But is there a way to acccess the cwd path which used to be `context.cwd`? Or maybe it's removed?

**Comment by meteorlxy** [comment, marked as answer]
The concept of `cwd` is only valid in cli. Core only need to handle absolute paths.

BTW, you'd better access those directory via `app.dir` helpers

**Reply by Mister-Hope** [reply]
The helpers are funtions with arguments, and I am not sure the meaning of those argument.

```ts
/**
 * Directory util function
 */
export type AppDirFunction = (...args: string[]) => string
```

Would it be fine for me to just call it?

Edit:

Are they paths to be joined?

```ts
/**
 * Directory util function
 */
export type AppDirFunction = (...paths: string[]) => string
```

I think rename the args would be better for understanding.

**Reply by meteorlxy** [reply]
Check the implementation and you'll easily understand it 😉

**Reply by Mister-Hope** [reply]
There is another thing I am concerned. I read though the source code, and I did not found any check about `base`. ( I am not prettey sure whether I miss anything)

Users should make sure `base` is starting and ending with a splash `/`, but some users may write like `base :'vuepress'`.

Vuepress V1 has special check with it, so it will work fine with config like 
https://github.com/vuejs/vuepress/blob/50388d9c009b5c5a16fd931c22d4ab59b4e00bbb/packages/%40vuepress/core/lib/node/__tests__/prepare/fixtures/docs-config/.vuepress/config.js#L1-L9 (V1 test, in fact most V1 test is just setting base to `vuepress`)

And it's safe for plugin and theme builders to access `context.base` without checking it.

Could you please add the checks in V2 too? So that we can just use `app.options.base`, thanks.
